### ISSUE TITLE
possibly null property not allowed by ibm-schema-type-format on OpenAPI 3.1

### ISSUE THREAD

**Issue by kornysietsma** [IBM/openapi-validator#748]
I'm trying to define a possibly null property, [as far as I can tell in OpenAPI 3.1 the syntax should be](https://stackoverflow.com/questions/23729973/how-to-specify-a-property-as-null-or-a-reference): 

```yaml
  myData:
    type: object
    properties:
        fooBar:
          anyOf:
            - $ref: '#/components/schemas/foos'
            - type: 'null'
```

This clashes with ibm-schema-type-format which returns an error:

```text
Message : Invalid type; valid types are: array, boolean, integer, number, object, string
```

**Comment by kornysietsma** [comment]
yep, trying to allow a null value like

```json
"myData":  {
    "fooBar": null
}
```

**Comment by m4rcingsxr** [comment]
Hello, also ibm-avoid-inline-schemas rule require null to be defined as $ref.
I don't know if this is a correct behaviour or not

**Comment by pyrooka** [comment]
@kornysietsma @m4rcingsxr We've just discussed this internally, but it will take some time to make progress. We'll post updates here.
